### PR TITLE
chore: Simplify artwork rail code II

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [ONYX-1290] 

### Description

This PR includes a few artwork rail code simplifications in preparation for the redesign.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Artwork rail code simplifications (refactoring) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1290]: https://artsyproduct.atlassian.net/browse/ONYX-1290?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f05f84c6f1b5e530cc11afe21952bf39301da32e
</p>

File: src/app/Components/ArtworkRail/ArtworkRail.tsx [hunk 1]
```diff
@@ -62,23 +59,8 @@ export const ArtworkRail: React.FC<ArtworkRailProps> = ({
...
-  const artworksSlice = artworks.slice(0, MAX_NUMBER_OF_ARTWORKS).map((artwork) => {
-    return {
-      ...artwork,
-      _disappearable: null,
-    }
-  })
-
-  const handleSupress = async (item: DissapearableArtwork) => {
-    await item._disappearable?.disappear()
-  }
```
Comment (olerichter00) [inline src/app/Components/ArtworkRail/ArtworkRail.tsx:80]: Moving all the disappearable logic to the rail card component.

File: src/app/Components/ArtworkRail/ArtworkRail.tsx [hunk 2]
```diff
@@ -88,43 +70,37 @@ export const ArtworkRail: React.FC<ArtworkRailProps> = ({
...
+            }}
+            showSaveIcon={showSaveIcon}
+            hideIncreasedInterestSignal={hideIncreasedInterestSignal}
+            hideCuratorsPickSignal={hideCuratorsPickSignal}
+            {...extractArtworkActionTrackingProps(otherProps)}
+          />
         )
       }}
-      keyExtractor={(item, index) => String(item.slug || index)}
+      keyExtractor={(item) => item.internalID}
```
Comment (olerichter00) [inline src/app/Components/ArtworkRail/ArtworkRail.tsx:103]: Using `internalID` because it's nonNullable.

File: src/app/Components/ArtworkRail/ArtworkRail.tsx [hunk 3]
```diff
@@ -88,43 +70,39 @@ export const ArtworkRail: React.FC<ArtworkRailProps> = ({
...
-                showSaveIcon={showSaveIcon}
-                onSupressArtwork={() => {
-                  handleSupress(item)
-                }}
-                hideIncreasedInterestSignal={hideIncreasedInterestSignal}
-                hideCuratorsPickSignal={hideCuratorsPickSignal}
-              />
-            </Box>
-          </Disappearable>
+          <Box pr="15px">
```
Comment (damassi) [inline src/app/Components/ArtworkRail/ArtworkRail.tsx:87]: Know its not part of the PR, but can we drop this line and stick to either `pr={1}` or `pr={2}` (design sys)?

Comment (olerichter00) [reply src/app/Components/ArtworkRail/ArtworkRail.tsx]: Very good idea 👍

Comment (olerichter00) [reply src/app/Components/ArtworkRail/ArtworkRail.tsx]: Changed it to `<Box pr={2}>` 🥈 🥈 🥈

**Review by MounirDhahri — APPROVED** [review]
LGTM! thanks for the refactor!
